### ISSUE TITLE
[Feature] Warn when tracing a model in train mode

### ISSUE THREAD

**Issue by TomSiegl** [sachinhosmani/torchvista#39]
Tracing a model in training mode can lead to internal updates in the model, e.g. in batch norm layers.
This side effect has the potential to impact model performance when working with pretrained weights and is likely not wanted.
On the other hand, tracing a model in training mode could be used to shed light on differences in the forward pass compared to eval mode.
Therefore I suggest to warn the user when tracing a model in training mode.

A minimal example:
```
import torch
import torchvista

bn = torch.nn.BatchNorm1d(1)
print(bn.running_mean, bn.running_var)

input = torch.randn(2, 1)
torchvista.trace_model(bn, input)
print(bn.running_mean, bn.running_var)
```
This outputs:
```
tensor([0.]) tensor([1.])
<IPython.core.display.HTML object>
tensor([0.0509]) tensor([0.9312])
```

**Comment by sachinhosmani** [comment]
Thanks for raising this @TomSiegl 

I had originally not thought of a reason why someone would use this tool in training mode. 

But I guess it can happen for two reasons:
1. The forward pass during train is actually designed to be different from during eval in the model, so it's actually worthwhile tracing it in train mode
2. The user forgets to ensure it's not in train mode

In both cases, I agree with your ask that we should warn the user like you suggested.

I will address this next.

**Comment by TomSiegl** [comment]
Thanks a lot for resolving this so quickly too!
